### PR TITLE
xbps-{create,rindex}: use zstd by default.

### PR DESCRIPTION
**Pull request by xtraeme**
There's lots of benchmarks right there, zlib vs zstd, xz vs zstd.
Take your own conclusion.

### CODE REVIEW

**Comment by xtraeme** [comment]
Closes #35

File: bin/xbps-create/main.c [hunk 1]
```diff
@@ -1050,10 +1050,14 @@ main(int argc, char **argv)
 	if (ar == NULL)
 		die("cannot create new archive");
 	/*
-	 * Set compression format, xz if unset.
+	 * Set compression format, zstd by default.
 	 */
-	if (compression == NULL || strcmp(compression, "xz") == 0) {
+	if (compression == NULL || strcmp(compression, "zstd") == 0) {
 		archive_write_add_filter_xz(ar);
```
Comment (Duncaen) [inline bin/xbps-create/main.c]: This is still `_xz`.

**Comment by Duncaen** [comment]
Not sure how I feel about this, how long has zstd been supported? What would be the timeframe in which a users who didn't update yet would have to do a update using static binaries?

File: bin/xbps-create/main.c (hunk 1 above)
Comment (xtraeme) [reply bin/xbps-create/main.c]: omg you are right, I forgot to change this, thanks

**Comment by Chocimier** [comment]
We could stay compatible by still using xz for index, xbps and all its dependencies - with compression per package option analogical to current build options per package.
Time of decompressing index is less than time of typing `xi -Su`. I do not think decreasing it is worth breaking updates.

File: bin/xbps-create/main.c [hunk 2]
```diff
@@ -1050,10 +1050,14 @@ main(int argc, char **argv)
...
+		archive_write_set_options(ar, "compression-level=19");
+	} else if (strcmp(compression, "xz") == 0) {
+		archive_write_add_filter_xz(ar);
+		archive_write_set_options(ar, "compression-level=19");
```
Comment (Chocimier) [inline bin/xbps-create/main.c]: Shouldn't be 9 for xz?

**Comment by xtraeme** [comment]
@Duncaen xbps supports reading zstd archives since 0.54.

**Comment by xtraeme** [comment]
so we can expect all users to have this version at least. Live images starting since 2019 should work with no issues.

**Comment by Duncaen** [comment]
This is only the latest ISO then, 0.54 was released 20190620.

I think I like the idea of having a `xbps-src` option to change the compressions for xbps and the packages xbps depends and keeping repodata as gzip for a few more 

I guess 6 months for rolling release is enough, everything older requiring manual work with static binaries isn't too bad.

**Comment by xtraeme** [comment]
`xbps-src` already supports this for a long time, I use:

```
[juan@leysa xbps]$ grep -i comp ~/.xbps-src.conf
XBPS_REPO_COMPTYPE=zstd
XBPS_PKG_COMPTYPE=zstd
[juan@leysa xbps]$
```

**Comment by Duncaen** [comment]
I mean use this on a per package basis if we would decide to not break compatibility with > 6 month outdated systems.

**Comment by xtraeme** [comment]
ah, sure.

**Comment by xtraeme** [comment]
hmm I think I'll add support to set compression level via environment. This way one could do:

```
$ COMPLEVEL=22 xbps-create ...
```

or something like this

**Comment by xtraeme** [comment]
Hmm, there's no >=0.54 available at https://a-hel-fi.m.voidlinux.org/static/

**Comment by xtraeme** [comment]
Can anyone please push 0.58?

**Comment by xtraeme** [comment]
This way we can make always a switch on old systems

**Comment by Duncaen** [comment]
There is 0.56 and 0.57, but I can make new ones.

**Comment by xtraeme** [comment]
oh true. Yes, please do. 0.58 is good enough

**Comment by xtraeme** [comment]
```
xbps-static-latest.aarch64-musl.tar.xz             12-Dec-2015 07:45             1679040
xbps-static-latest.armv6l-musl.tar.xz              12-Dec-2015 07:45             1613140
xbps-static-latest.i686-musl.tar.xz                12-Dec-2015 07:45             2281348
xbps-static-latest.mips-musl.tar.xz                12-Dec-2015 07:45             2324644
xbps-static-latest.mipsel-musl.tar.xz              12-Dec-2015 07:45             2454672
xbps-static-latest.x86_64-musl.tar.xz              12-Dec-2015 07:45             2105416
```

please point latest to latest :-)

**Comment by Duncaen** [comment]
I do think https://github.com/void-linux/xbps/issues/234 should be fixed with with a patch in void-packages, before we make a new release.

**Comment by Duncaen** [comment]
Updated the static builds for the architectures we have, i686 is missing because we don't build the musl package.

**Comment by xtraeme** [comment]
@Duncaen thanks!

**Pull request merged**